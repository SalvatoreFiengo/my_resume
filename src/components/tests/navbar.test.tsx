/**
 * @jest-environment jsdom
 */

import { shallow, render, mount } from '../../enzyme';
import Navbar from '../header/navbar';

type TestProps = {
    cvPosition: number;
    skillPosition: number;
    workHistoryPos:number; 
    educationPos:number
};

describe('navbar', ()=>{
    it('renders without crashing',()=>{
        const props:TestProps = {cvPosition: 60, skillPosition: 80, workHistoryPos: 100, educationPos: 120};
        const navbarComponent = shallow(<Navbar {...props}/>);
        expect(navbarComponent.getElements()).toMatchSnapshot();
    })
    it('renders navlist with class d-none in screen sizes over medium breakpoint, d-block when burger menu is clicked',()=>{
        const props:TestProps = {cvPosition: 60, skillPosition: 80, workHistoryPos: 100, educationPos: 120};
        const navbarComponent = mount(<Navbar {...props}/>);
        const navBurgerMenu = navbarComponent.find('button.border');
        const listIsHidden = navbarComponent.find('ul.d-none');
        // before click      
        expect(listIsHidden).toHaveLength(1);
        
        global.innerWidth = 300;
        expect(navBurgerMenu).toHaveLength(1);
        navBurgerMenu.simulate('click');
        // after click
        const listIsShowing = navbarComponent.find('ul.d-block');
        expect(listIsShowing).toHaveLength(1)
        navbarComponent.unmount();
    })
    it('scrolls to a specific seciton when nav button is clicked',()=>{ 
        window.scrollTo = jest.fn();
        const props:TestProps = {cvPosition: 60, skillPosition: 80, workHistoryPos: 100, educationPos: 120};
        const navbarComponent = mount(<Navbar {...props}/>);

        const presentationButton = navbarComponent.findWhere(node=>node.type()==='button' && node.text()==='Presentation');
        presentationButton.simulate('click');
        expect(window.scrollTo).toBeCalledWith(0,0);

        const SkillsButton = navbarComponent.findWhere(node=>node.type()==='button' && node.text()==='Skills');
        SkillsButton.simulate('click');
        expect(window.scrollTo).toBeCalledWith(0,20);

        const EducationButton = navbarComponent.findWhere(node=>node.type()==='button' && node.text()==='Education');
        EducationButton.simulate('click');
        expect(window.scrollTo).toBeCalledWith(0,60);

        const WorkHistoryButton = navbarComponent.findWhere(node=>node.type()==='button' && node.text()==='Work History');
        WorkHistoryButton.simulate('click');
        expect(window.scrollTo).toBeCalledWith(0,40);

        navbarComponent.unmount();
        jest.clearAllMocks();
    })
})