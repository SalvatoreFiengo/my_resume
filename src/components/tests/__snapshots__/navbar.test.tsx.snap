// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`navbar renders without crashing 1`] = `
Array [
  <styled.nav
    className="w-100 pt-2 nav-relative bg-transparent"
  >
    <div>
      <div
        className="row"
      >
        <div
          className="col-4 col-md-3"
        >
          <styled.button
            className="navbar-brand ms-3 ms-md-5 pt-0"
            onClick={[Function]}
          >
            My Resume
          </styled.button>
        </div>
        <div
          className="d-block d-md-none col-8 text-end"
        >
          <button
            className="border mb-1 me-5"
            onClick={[Function]}
            type="button"
          >
            <i
              aria-hidden="true"
              className="fa fa-2x fa-bars"
            />
          </button>
        </div>
        <div
          className="col-4 col-md-7"
        >
          <styled.ul
            className="d-md-inline mr-3 mt-md-2 mb-1 d-none"
          >
            <li
              className="d-md-inline me-md-3"
            >
              <styled.button
                onClick={[Function]}
              >
                Presentation
              </styled.button>
            </li>
            <li
              className="d-md-inline me-md-3"
            >
              <styled.button
                onClick={[Function]}
              >
                Skills
              </styled.button>
            </li>
            <li
              className="d-md-inline me-md-3"
            >
              <styled.button
                onClick={[Function]}
              >
                Education
              </styled.button>
            </li>
            <li
              className="d-md-inline"
            >
              <styled.button
                onClick={[Function]}
              >
                Work History
              </styled.button>
            </li>
          </styled.ul>
        </div>
      </div>
    </div>
  </styled.nav>,
]
`;
